kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/filter/NettyFilterDispatchExecutorTest.java

```diff
@@ -0,0 +1,347 @@
+            assertThat(dispatchExecutor.getEventLoop()).isSameAs(eventLoop);
+        }
+        finally {
+            eventLoop.shutdownGracefully();
+        }
+    }
+
+    @Test
+    void notInFilterDispatchThreadWhenInTestThread() {
+        EventLoop eventLoop = new DefaultEventLoop();
+        try {
+            NettyFilterDispatchExecutor dispatchExecutor = new NettyFilterDispatchExecutor(eventLoop);
+            assertThat(dispatchExecutor.isInFilterDispatchThread()).isFalse();
+        }
+        finally {
+            eventLoop.shutdownGracefully();
+        }
+    }
+
+    @Test
+    void notInFilterDispatchThreadWhenInAnotherEventLoop() {
+        EventLoop eventLoop = new DefaultEventLoop();
+        EventLoop anotherEventLoop = new DefaultEventLoop();
+        try {
+            NettyFilterDispatchExecutor dispatchExecutor = new NettyFilterDispatchExecutor(eventLoop);
+            Future<Boolean> inDispatchThreadFuture = anotherEventLoop.submit(dispatchExecutor::isInFilterDispatchThread);
+            assertThat(inDispatchThreadFuture).succeedsWithin(5, TimeUnit.SECONDS).isEqualTo(false);
+        }
+        finally {
+            eventLoop.shutdownGracefully();
+            anotherEventLoop.shutdownGracefully();
+        }
+    }
+
+    @Test
+    void inFilterDispatchThreadWhenInEventLoop() {
+        EventLoop eventLoop = new DefaultEventLoop();
+        try {
+            NettyFilterDispatchExecutor dispatchExecutor = new NettyFilterDispatchExecutor(eventLoop);
+            Future<Boolean> inDispatchThreadFuture = eventLoop.submit(dispatchExecutor::isInFilterDispatchThread);
+            assertThat(inDispatchThreadFuture).succeedsWithin(5, TimeUnit.SECONDS).isEqualTo(true);
+        }
+        finally {
+            eventLoop.shutdownGracefully();
+        }
+    }
+
+    @Test
+    void inFilterDispatchThreadByDelegation() {
+        EventLoop eventLoop = new DefaultEventLoop();
+        try {
+            NettyFilterDispatchExecutor dispatchExecutor = new NettyFilterDispatchExecutor(eventLoop);
+            Future<Boolean> inDispatchThreadFuture = dispatchExecutor.submit(dispatchExecutor::isInFilterDispatchThread);
+            assertThat(inDispatchThreadFuture).succeedsWithin(5, TimeUnit.SECONDS).isEqualTo(true);
+        }
+        finally {
+            eventLoop.shutdownGracefully();
+        }
+    }
+
+    @Test
+    void switchToFilterDispatchThreadWhenInEventLoop() {
+        EventLoop eventLoop = new DefaultEventLoop();
+        try {
+            NettyFilterDispatchExecutor dispatchExecutor = new NettyFilterDispatchExecutor(eventLoop);
+            CompletableFuture<Object> completionStage = new CompletableFuture<>();
+            CompletionStage<Object> objectCompletionStage = dispatchExecutor.switchToFilterDispatchThread(completionStage);
+            CompletionStage<Boolean> inDispatchThread = objectCompletionStage.thenApply(o -> dispatchExecutor.isInFilterDispatchThread());
```

Comment:
tempted to inline `objectCompletionStage`
```suggestion
            CompletionStage<Boolean> inDispatchThread = dispatchExecutor.switchToFilterDispatchThread(completionStage)
            .thenApply(o -> dispatchExecutor.isInFilterDispatchThread());
```